Implemented the advanced search header for the home page

Julyuk opened the pull request.

Julyuk committed:
Implemented the advanced search header for the home page

s-hrynus commented on src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java:
```diff
@@ -0,0 +1,70 @@
+
+    public WebElement getAdvancedSearchTextHeading() {
+        if (advancedSearchTextHeading == null) {
+            advancedSearchTextHeading = rootElement.findElement(By.xpath("//div[contains(@class, \"city-name-box\")]/h2"));
```
```suggestion
            advancedSearchTextHeading = rootElement.findElement(By.xpath("//h2[@class='city-name']"));
```
You can use a straight path

s-hrynus commented on src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java:
```diff
@@ -0,0 +1,70 @@
+    }
+    public WebElement getSelectionSearchInputFieldPlaceholder() {
+        if (selectionSearchInputFieldPlaceholder == null) {
+            selectionSearchInputFieldPlaceholder = rootElement.findElement(By.xpath("//div[contains(@class, \"search\")]//span[contains(@class,\"ant-select-selection-placeholder\")]"));
```
```suggestion
            selectionSearchInputFieldPlaceholder = rootElement.findElement(By.xpath("//span[@class='ant-select-selection-placeholder']"));
```

Same

s-hrynus reviewed (approved):
Great job 👍

Julyuk committed:
Update src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java

improved the xpath expression for the advancedSearchTextHeading according to the recommendation

Co-authored-by: Hrynus Stanislav <69212372+s-hrynus@users.noreply.github.com>

Julyuk committed:
Update src/main/java/com/academy/ui/pages/advancedSearchHeader/AdvancedSearchHeaderComponent.java

Improved the xpath expression for the selectionSearchInputFieldPlaceholder according to the recommendation

Co-authored-by: Hrynus Stanislav <69212372+s-hrynus@users.noreply.github.com>

Julyuk committed:
Added methods for setting the text for the input field and checking it

Julyuk committed:
Removed an excessive method

lhalam commented on src/main/java/com/academy/ui/pages/BasePage.java:
```diff
@@ -20,6 +23,14 @@ public HeaderComponent getHeader() {
+        }
+        return advancedSearchHeaderComponent;
+    }
+
```
create page BasePageWithAdvancedSearch
move impl to BasePageWithAdvancedSearch

Julyuk committed:
Moved the advancedSearch component to a separate base page with advancedSearch

lhalam committed:
Merge branch 'develop' into julyuk-implement-advancedsearchheadercomponent
